scripts-available/CDB_FederatedServerListRemote.sql

```diff
@@ -0,0 +1,299 @@
+--------------------------------------------------------------------------------
+-- Private functions
+--------------------------------------------------------------------------------
+
+--
+-- List the schemas of a remote PG server
+-- 
+CREATE OR REPLACE FUNCTION @extschema@.__CDB_FS_List_Foreign_Schemas_PG(server_internal name)
+RETURNS TABLE(remote_schema name)
+AS $$
+DECLARE
+    -- Import schemata from the information schema
+    --
+    -- "The view schemata contains all schemas in the current database
+    -- that the current user has access to (by way of being the owner
+    -- or having some privilege)."
+    -- See https://www.postgresql.org/docs/11/infoschema-schemata.html
+    --
+    -- "The information schema is defined in the SQL standard and can
+    -- therefore be expected to be portable and remain stable"
+    -- See https://www.postgresql.org/docs/11/information-schema.html
+    inf_schema name := 'information_schema';
+    remote_table name := 'schemata';
+    local_schema name := @extschema@.__CDB_FS_Create_Schema(server_internal, inf_schema);
+    role_name name := @extschema@.__CDB_FS_Generate_Server_Role_Name(server_internal);
+BEGIN
+    -- Import the foreign schemata table
+    IF NOT EXISTS (
```

Comment:
I think there's a common pattern here that could be refactored. In pseudo code: `IF NOT EXISTS table/schema THEN CREATE/IMPORT table/schema`.

BTW: you may want to check the target schema exists before

Reply:
The target schema is always created, if it doesn't already, exist with:
```
local_schema name := @extschema@.__CDB_FS_Create_Schema(server_internal, inf_schema);
```